### DISCUSSION TITLE
0610 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#106] [General]
# 오늘의 이모지

| 아라       | 재훈       | 유라       | 민용       | 정현       | 진현       |
| ---------- | ---------- | ---------- | ---------- | ---------- | ---------- |
| :question: | :question: | :question: | :question: | :question: | :question: |

# 회의 및 토론

**[재훈]** **pycaret tabular.py 함수수정**

train_test_split()의 ratio와 상관 없이 데이터 전체를 가져옴

**pycaret process**의 create_model을 할때부터 1.0을 다쓰도록 하였다. validation 기준이 생김 (과적합이 있을 수는 있음)

**[미뇽]** **riiid 데이터로 pretrain 시킨 모델을 가지고 학습완료**
= 효과가 그렇게 있지 않았다. 
= Embedding 레이어를 새로 학습하고 GRU랑 Encoding Layer만 가져오다 보니까 영향이 크진 않았다.

**onehotencoding** 
Embedding layer거치고 padding 을 주면 (0, 1, 2) 사이의 값으로 encoding 됨
= valid_auc는 떨어졌는데 LB 조금 오름
= 낮았던 fold에서 상승함으로써 각 fold가 적어도 .805 이상이 되었다.
= ml 도 이용했으면 좋겠다. 

**(피쳐)** problem_number / 푼시간 
= 구간별로 나누어서 적용함 (30구간) 
= 50개정도 추가가 됨. 

정답률을 이용해서 stratefied에 들어갈 key를 10개로 구성해서 진행중이다. 

**[진현]**  **SAINT에서 LB - Valid_Auc = 0.015** 

LB는 0.78정도 예상 _ 진현피처 추가 | 직접 베이스라인 구축해서 해봄

assessmentItem_mean과 관련해서는 다 해봤는데 빼고 넣었을 때 가장 잘 나옴!

SAINT+를 해보고 싶어서 논문을 읽고 적용을 하는 중

**[재훈]**

민용이가 썼던 검증전략을 가져와도 모델이 달라지니 다르게 나왔다. 떨어지거나 그대로.

( 모델에 따라서 다른 검증전략을 써야될 수 도 있어 _ 아라 )

( 모델의 표현력이 낮아서 그럴 수 있음 0.815이전까지는 비슷하게 가져갔지만 그 이후로 달라질 수도 있음 _ 민용 )

**Validation** 1997 seed stratfied => 마지막문제를 맞춘 userId로 기준을 세우기

( stratified를 나누되 실험은 single fold로 해보면 어떨까 )
continuous 와 categorical 을같이 넣어주니까 하나가 f impor가 0이 나옴 

(cate만 써보기도 해야겠다 _ 민용)

**[아라]** **Last query 하이퍼파라미터 차이: 0.0439** 

strkfold 적용: Valid AUC: 0.8191 LB: 0.7970 (차이: 0.0221)

random으로 하거나 Feature를 고득점과 비슷하게 넣어주고 있는데 나아지진 않았다.

AUC  = (FPR , TPR) 기준으로 좀 더 살펴봄.

pretrain을 시켜서 학습하는 지점이 차이가 많이 나는 부분이었다.
유지할지는 모르겠지만 validation전략을 먼저 잡는게 나을거 같다.

**[ 1d Convolution ]** 

​	conv레이어로 특징을 뽑고 avgpooling 다음에 fully connected layer을  통과시켜보려고 한다. (encoder안붙이고 따로 해볼 예정)

​	n = 250, k = (feature들의 dimension) 3으로, 시험지 하나당 문제 개수가 13개 혹은 그 이내의 수로 해볼것

​	convolutional Layer를 세개 정도 올려볼 예정 

​	**regression**도 시도해봄 valid_auc 0.68xx

​	attention을 avg pooling에 넣어봄

[문제 풀이 시간 관련]

300초 기준으로 60x60x24 or 60x60x12 해봤는데 5분으로 해보고 분단위로 접근해보려고 한다.
( 5분이 1, 2번문제 풀고 3번 문제를 4시간 뒤에 푸는 애들도 있기에 300초로 해보긴 했는데 바꿔서 생각하는 것도 괜찮은것 같다 _ 민용)

**[정현]** LSTMATTN 결과 : 0.7 중반 (민용피처 이용, hyperparameter 초기값)

Hyperparameter를 결과가 좋은 걸로 다시 이어서 실험해보기. pretrain까지 이어서 진행해볼 예정.

**[유라]**

* kfold 다시 만들고 나서, 조교님이 언급하셨던 그래프 갖구 피처 다루는 부분 찾아봄
*  knowledgeTag 사이의 관계를 임베딩 하는 방식 읽어봄. DL쪽이고 SAINT결과보다 점수가 좋지 않아서 적용하기는 힘들듯!
* 그래프로 feature selection 하는 글 체크. feature가 많이 필요할 거 같음
* cont feature 관련 interaction period와 Minmaxscaler 후처리 진행 + 성능 0.813 정도
  (minmax scaler는 Caret (Tree 기반) 모델에서는 별로 효과가 없다고 알고있어서 유의할지 잘 모르겠당 _ 재훈) 
* feature 잔뜩 만들고 PCA나 LDA 적용해서 또 feature 뽑구 하려고 함.
* holdout 구성할 때 테스트 셋의 맨 마지막에서 두 번째 row 그니까 맞추기 직전 것을 holdout으로 써볼까 함.
  (fix_imbalance  옵션 추가 고려중_ 재훈) 
  ( test set에서 holdout 구성하고 train할 때는 train만 써두 되니까 train_except_holdout: False => True로 바꿔보는것도 좋을듯 )

진현피처 추가 질문 => 300기준으로 자르고 나머지는 301
근데 300에 걸치는 건 별로 없고 300 보다 큰건 18% 정도!

**제출 횟수 계획**

**민용 2 재훈 2+ 진현 1 정현 1 아라 2+**

**발표 계획**

발표 자료 작성 => 진현!

1. 관점별로 나눠서 소개하기 

2. 각자가 도움이 되는 걸 떠올려오기

3. [발표자] 해보고나서 가장 깔끔한사람이 하는 걸로

ex) hold - out을 넣는다면 validation기준으로 ! 키워드 3개정도 준비해서_ 대분류, 중분류, 소분류 및 이유까지 준비해오자!

# 저녁메뉴

| 아라       | 재훈       | 유라       | 민용       | 정현       | 진현       |
| ---------- | ---------- | ---------- | ---------- | ---------- | ---------- |
| :question: | :question: | :question: | :question: | :question: | :question: |

**Comment by MoonJaeHoon** [comment]
형님 항상 감사링 ㅎㅎ

**Reply by MoonJaeHoon** [reply]
# 오늘할일

1. Validation Set 구축 여러가지 실험해봐야함. (Generalized를 위해서)
  - Permutation Importance도 적용하려면 믿을만한 Valid Set이 필요하다
2. Window Size를 고려하여 변수들 증식 시도해볼 것임.

**Comment by arabae** [comment]
- [x] CNN 모델 만들어서 실험
- [x] CNN 깊게 + skip connection
- [x] GAP -> SAP (self-attentive pooling)
- [x] Feature 변경 이야기
- [ ] CNN hyperparameter 찾기
